bootloader/src/pyi_win32_utils.c

```diff
@@ -435,6 +435,55 @@ pyi_win32_utf8_to_mbs(char * dst, const char * src, size_t max)
     }
 }
 
+
+/* Retrieve the SID of the current user.
+ *  Returns SID string on success, NULL on failure.
+ */
+static char *
+pyi_win32_get_user_sid()
+{
+    HANDLE process_token;
+    DWORD user_info_size, result;
+    PTOKEN_USER user_info;
+    wchar_t *wsid = NULL;
+    char *sid;
+
+    // Get access token for the calling process
+    if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &process_token)) {
+        return NULL;
+    }
+
+    // Get buffer size and allocate buffer
+    if (!GetTokenInformation(process_token, TokenUser, NULL, 0, &user_info_size)) {
+        if (GetLastError() != ERROR_INSUFFICIENT_BUFFER) {
```

Comment:
Why is ERROR_INSUFFICIENT_BUFFER not treated like all other errors?

Reply:
Because this call is used as a query for the required buffer size. We pass NULL and 0 for the buffer and its size, and expect the call to return ERROR_INSUFFICIENT_BUFFER and store the required buffer size to user_info_size. Then we allocate the buffer, and repeat the call with actual buffer and its size.

Reply:
IC, as the comment above says :-)